src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -489,26 +490,30 @@ public void AddNonLocals(DocumentId documentId, ImmutableArray<DiagnosticData> d
                     Add(ref _nonLocals, documentId, diagnostics);
                 }
 
-                public void AddOthers(ProjectId unused, ImmutableArray<DiagnosticData> diagnostics)
+                public void AddOthers(ImmutableArray<DiagnosticData> diagnostics)
                 {
                     _others = diagnostics;
                 }
 
                 private void Add(ref ImmutableDictionary<DocumentId, ImmutableArray<DiagnosticData>>.Builder locals, DocumentId documentId, ImmutableArray<DiagnosticData> diagnostics)
                 {
+                    if (_project.GetDocument(documentId)?.SupportDiagnostics() == false)
```

Comment:
what's the case where this is referrign to a document that isn't in the project anymore?  This seems racy...

Reply:
build